hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/AbstractTestS3AEncryption.java

```diff
@@ -107,10 +106,15 @@ public void testEncryptionOverRename() throws Throwable {
     validateEncrytionSecrets(secrets);
     writeDataset(fs, src, data, data.length, 1024 * 1024, true);
     ContractTestUtils.verifyFileContents(fs, src, data);
-    Path dest = path(src.getName() + "-copy");
-    fs.rename(src, dest);
-    ContractTestUtils.verifyFileContents(fs, dest, data);
-    assertEncrypted(dest);
+    Path targetDir = path("target");
```

Comment:
I am looking at this, trying to understand what it is doing.

Before: we created a file src, renamed it to dest and verified the contents were unchanged; dest encrypted.

After: 
1. src is created as a dataset
1. new path targetDir created
1. file `dest` is created in a target/src+"-another" with a different dataset; contents verified
1. rename(src, targetDir) to create the file targetDir/src
1. which is verified
1. dest file is completely ignored

So why the change here? I don't see why we need the new test file, and the only change is now that you're renaming into a subdirectory  which already exists rather than a path of the destination file.

I need some clarification here.
* why the change
* before the encryption settings were changed in copy, how did this new test fail?

Reply:
> 6. dest file is completely ignored
> 
The reason dest file has to be created is to enforce rename to consider targetDir as a directory else it considers it as file. 

> I need some clarification here.
> 
> * why the change
>
This change was done to address one of your above comment.
"Maybe: in testEncryptionOverRename , rename the file into a directory." 

> * before the encryption settings were changed in copy, how did this new test fail?
> 
The encryption key of the destination file  targetDir/src was not matching with the configured kms key of the bucket rather it was equal to some default key generated by the S3 itself.

Reply:
bq. The reason dest file has to be created is to enforce rename to consider targetDir as a directory else it considers it as file.

mkdir(targetDir) should have done that. Or is it not because of that funny "rename into empty dir" problem with rename() which everyone hates (historical mistake, BTW)

If somehow that doesn't work and you want to create a file, ContractTestUtils.touch() will do that; add a comment above about why its needed